common/Services/NotificationService.cs

```diff
@@ -0,0 +1,214 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using CommunityToolkit.Mvvm.Input;
+using CommunityToolkit.WinUI;
+using CommunityToolkit.WinUI.Behaviors;
+using DevHome.Common.Contracts;
+using DevHome.Common.Environments.Helpers;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.Windows.AppLifecycle;
+using Microsoft.Windows.AppNotifications;
+using Microsoft.Windows.AppNotifications.Builder;
+using Serilog;
+using Windows.ApplicationModel.Activation;
+
+namespace DevHome.Common.Services;
+
+public class NotificationService
+{
+    private readonly ILogger _log = Log.ForContext("SourceContext", nameof(NotificationService));
+
+    private readonly IWindowsIdentityService _windowsIdentityService;
+
+    private readonly string _componentName = "NotificationService";
+
+    private readonly StringResource _stringResource;
+
+    private readonly Microsoft.UI.Dispatching.DispatcherQueue _dispatcher;
+
+    private StackedNotificationsBehavior? _notificationQueue;
+
+    public NotificationService(IWindowsIdentityService windowsIdentityService)
+    {
+        _windowsIdentityService = windowsIdentityService;
+        _dispatcher = Microsoft.UI.Dispatching.DispatcherQueue.GetForCurrentThread();
+        _stringResource = new StringResource("DevHome.Common.pri", "DevHome.Common/Resources");
+    }
+
+    public bool ShowHyperVAdminWarningToast()
+    {
+        // Temporary toast notification to inform the user that they are not in the Hyper-V admin group.
+        // In the future we'll use an admin process from Dev Home to add the user to the group.
+        var toast = new AppNotificationBuilder()
+           .AddText("Warning")
+           .AddText(StringResourceHelper.GetResource(StringResourceHelper.UserNotInHyperAdminGroupMessage))
+           .AddButton(new AppNotificationButton(StringResourceHelper.GetResource(StringResourceHelper.UserNotInHyperAdminGroupButton))
+           .AddArgument("action", "AddUserToHyperVAdminGroup"))
+           .BuildNotification();
+
+        AppNotificationManager.Default.Show(toast);
+        return toast.Id != 0;
+    }
+
+    public void Initialize(StackedNotificationsBehavior notificationQueue)
+    {
+        _notificationQueue = notificationQueue;
+    }
+
+    public void HandlerNotificationActions(AppActivationArguments args)
+    {
+        if (args.Data is ToastNotificationActivatedEventArgs toastArgs)
+        {
+            try
+            {
+                if (toastArgs.Argument.Contains("action=AddUserToHyperVAdminGroup"))
+                {
+                    // Launch compmgmt.msc in powershell
+                    var psi = new ProcessStartInfo();
```

Comment:
How do you uninstall PowerShell? AFAIK it's a Windows component and uninstalling it is not a supported scenario on Windows (this may change in the future, but not any time soon). If you are doing something that OS doesn't support you are on your own.